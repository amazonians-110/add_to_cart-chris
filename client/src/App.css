.container{
	position : relative;
	top : 100px;
	display : flex;
	width : 200px;
	height : 500px; 
	flex-direction : column;
	background-color : white;
	float : right;
	border: 1px #D3D3D3 solid;
	padding: 10px 20px; 

}

.price{
	color : #B12704;
	font-size: 17px;
}

.freeshipping{
	margin-top : 10px;
	font-size : 13px;
	line-height : 1.6;
}


.details{
   color :#0066c0;
}

.details:hover{
  color :  #B12704;
  cursor : pointer;
}

