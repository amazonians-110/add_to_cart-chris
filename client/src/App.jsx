import React from 'react';
import CSSModules from 'react-css-modules';
import './App.css';
import axios from 'axios';
import DetailsBox from './DetailsBox.jsx';
import Dates from './Dates.jsx';
import Buttons from './Buttons.jsx';
import DeliverModal from './DeliverModal.jsx';
import AddToListModal from './AddToListModal.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      price: ''
    };
    this.child = React.createRef();
    this.clickOutside = this.clickOutside.bind(this);
  }


  componentDidMount() {
    const urlArray = document.URL.split('/');
    const productID = Number(urlArray[urlArray.length - 1]);
    axios.get(`http://127.0.0.1:3002/api/product/${productID}`)
      .then((res) => {
        this.setState({ price: res.data[0].price });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  clickOutside(event) {
    if (event.target.className === 'body' || event.target.className === 'container') {
      this.child.current.setState({ clicked: false });
    }
  }

  render() {
    return (
      <div className="body" onClick={this.clickOutside}>
        <div className="container">
          <div className="price"> ${this.state.price} </div>
          <div className="freeshipping">
            &
            {' '}
            <b>FREE Shipping</b>
            {' '}
            on orders over $25 shipped by Amazon.
            {' '}
            <DetailsBox ref={this.child} />
            <Dates />
            <Buttons />
            <DeliverModal />
            <AddToListModal />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
