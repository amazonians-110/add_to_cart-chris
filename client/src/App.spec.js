import React from 'react';
import { shallow } from 'enzyme';
import App from './App.jsx';

describe('App', () => {
  it('should render', () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });
});
