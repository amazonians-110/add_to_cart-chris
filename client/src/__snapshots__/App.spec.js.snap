// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render 1`] = `
<div
  className="body"
  onClick={[Function]}
>
  <div
    className="container"
  >
    <div
      className="price"
    >
       $19.99 
    </div>
    <div
      className="freeshipping"
    >
      &
       
      <b>
        FREE Shipping
      </b>
       
      on orders over $25 shipped by Amazon.
       
      <DetailsBox />
      <Dates />
    </div>
  </div>
</div>
`;
